src/EPR.Calculator.API/Controllers/CalculatorController.cs

```diff
@@ -52,5 +56,30 @@ public IActionResult GetCalculatorRun()
             // TODO: Return the details of a particular run
             return new OkResult();
         }
+
+        [HttpGet]
+        [Route("calculatorRunByNameExist/{name}")]
+        public IActionResult GetCalculatorRunByName([FromRoute] string name)
+        {
+            if (!ModelState.IsValid)
+            {
+                return StatusCode(StatusCodes.Status400BadRequest, ModelState.Values.SelectMany(x => x.Errors));
+            }
+
+            try
+            {
+                var calculatorRun = _context.CalculatorRuns.AsEnumerable().Any(run => string.Equals(run.Name, name, StringComparison.OrdinalIgnoreCase));
```

Comment:
We dnt need AsEnumerable() here. 
In fact my previous comment I have given you the code - https://github.com/DEFRA/epr-calculator-api/pull/53#discussion_r1793299755

Reply:
Hi Uday Previously I have implemented that but that code was showing i issue on Sonar and was refereeing to the solution to use the StringComparision .

Reply:
In regards to function i have tried use the same approach but with .any() but then it throw an exception so i have use the  AsEnumerable() method that switches the query from being evaluated on the database server to being evaluated in memory on the client side in case that are not supported by the database provider.